cpp/benchmarks/common/generate_benchmark_input.hpp

```diff
@@ -1,248 +1,390 @@
-#pragma once
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 
-#include <cudf/column/column.hpp>
-#include <cudf/table/table.hpp>
+#pragma once
 
-#include <tests/utilities/column_utilities.hpp>
-#include <tests/utilities/column_wrapper.hpp>
+#include <map>
 
-#include <memory>
-#include <random>
-#include <vector>
+#include <cudf/table/table.hpp>
+#include <cudf/utilities/traits.hpp>
 
 /**
  * @file generate_benchmark_input.hpp
- * @brief Contains functions that generate columns filled with random data.
+ * @brief Contains declarations of functions that generate columns filled with random data.
  *
- * Also includes utilies that generate random tables.
+ * Also includes the data profile descriptor classes.
  *
- * The distribution of random data is meant to simulate real-world data. For example, numerical
- * values are generated using a normal distribution with a zero mean. Therefore, different column
- * types are filled using different distributions. The distributions are documented in the
- * functions where they are used.
+ * The create_random_table functions take a data profile, the information about table size and a
+ * seed to deterministically generate a table with given parameters.
  *
  * Currently, the data generation is done on the CPU and the data is then copied to the device
  * memory.
  */
 
 /**
- * @brief Mersenne Twister engine with static seed.
- *
- * Produces the same random sequence on each run.
+ * @brief Identifies a group of related column's logical element types.
  */
-auto& deterministic_engine()
-{
-  static unsigned seed = 13377331;
-  static std::mt19937 engine{seed};
-  return engine;
-}
+enum class type_group_id : int32_t {
+  INTEGRAL = static_cast<int32_t>(cudf::type_id::NUM_TYPE_IDS),
+  INTEGRAL_SIGNED,
+  FLOATING_POINT,
+  NUMERIC,
+  TIMESTAMP,
+  DURATION,
+  FIXED_POINT,
+  COMPOUND,
+  NESTED,
+};
 
 /**
- * @brief Type trait for cudf timestamp types.
+ * @brief Identifies a probability distribution type.
  */
-template <typename>
-struct is_timestamp {
-  constexpr static bool value = false;
+enum class distribution_id : int8_t {
```

Comment:
Could we get comments with some qualitative descriptions of these distributions?  Maybe some examples of why you might want to choose each one.